components/PostingLocation/SuggestionsRooms/SuggestionsRooms.jsx

```diff
@@ -109,6 +128,18 @@ export class SuggestionsRooms extends Component {
     if (!this.state.listening) {
       this.addInputListener();
     }
+    //Remove rooms already filtered from suggestions
+    let currentFilters = this.state.filters;
+    for(let i = 0, n = suggestionsList.length; i < n; i++) {
+      for(let k = 0, l = currentFilters.length; k < l; k++) {
```

Comment:
Same comment above.

You might create a hashtable from currentFilters array. Besides that you are able to remove duplications from suggestionsList array.

This approach can simplify the code, has a great performance and use immutable arrays.